configure.ac

```diff
@@ -611,33 +611,50 @@ else
   AC_SUBST([HAS_SSH_LOCALHOST], [no])
 fi
 
-#check for mpich2
+#check for mpich3
 AC_ARG_WITH([mpich],
-            [AS_HELP_STRING([--with-mpich=path],
-                            [Compile mpich2 tests in `make check`])],
-            [mpich=$withval],
-            [mpich=no])
-
-if test "$mpich" != "no"; then
-  mpich_path="$mpich:$mpich/bin"
-  AC_PATH_PROG([MPICH_MPD],        [mpd],        [no], [$mpich_path])
-  AC_PATH_PROG([MPICH_MPDBOOT],    [mpdboot],    [no], [$mpich_path])
-  AC_PATH_PROG([MPICH_MPDALLEXIT], [mpdallexit], [no], [$mpich_path])
-  AC_PATH_PROG([MPICH_MPDCLEANUP], [mpdcleanup], [no], [$mpich_path])
-  AC_PATH_PROG([MPICH_MPICC],      [mpicc],      [no], [$mpich_path])
-  AC_PATH_PROG([MPICH_MPIEXEC],    [mpiexec],    [no], [$mpich_path])
-  if test "$MPICH_MPD" = "no"        ||\
-     test "$MPICH_MPDBOOT" = "no"    ||\
-     test "$MPICH_MPDALLEXIT" = "no" ||\
-     test "$MPICH_MPDCLEANUP" = "no" ||\
+            [AS_HELP_STRING([--with-mpich=mpich_path_prefix],
+                            [Compile mpich3 tests in `make check`])],
+            [mpich_path_prefix=$withval],
+            [mpich_path_prefix=no])
+
+mpich_path="no"
+if test "$mpich_path_prefix" != "no"; then
+  mpich_path="$mpich_path_prefix:$mpich_path_prefix/bin"
+else
+  # Let's see if we can find mpicc in the default path
+  mpiccpath=`which mpicc`
```

Comment:
Hi @gc00, should we just set `mpiccpath` to `$PATH` (or the autoconf equivalent) instead? I vaguely remember debian or some other place frowning upon use of `which` to find executables.

If we don't find mpich in the user-path or system-path, we can simply say `not found`. Is this also your goal to find if the user gave us wrong mpich path?